cli/root.go

```diff
@@ -4,64 +4,77 @@ import (
 	"net/url"
 	"os"
 
-	"github.com/alecthomas/kingpin"
-	"github.com/prometheus/alertmanager/cli/format"
 	"github.com/prometheus/common/version"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	"github.com/prometheus/alertmanager/cli/config"
+	"github.com/prometheus/alertmanager/cli/format"
 )
 
 var (
 	verbose         bool
 	alertmanagerURL *url.URL
 	output          string
+
+	configFiles = []string{os.ExpandEnv("$HOME/.config/amtool/config.yml"), "/etc/amtool/config.yml"}
+	legacyFlags = map[string]string{"comment_required": "require-comment"}
 )
 
+func requireAlertManagerURL(pc *kingpin.ParseContext) error {
+	// Return without error if any help flag is set.
+	for _, elem := range pc.Elements {
+		f, ok := elem.Clause.(*kingpin.FlagClause)
+		if !ok {
+			continue
+		}
+		name := f.Model().Name
+		if name == "help" || name == "help-long" || name == "help-man" {
+			return nil
+		}
+	}
+	if alertmanagerURL == nil {
+		kingpin.Fatalf("required flag --alertmanager.url not provided")
+	}
+	return nil
+}
+
 func Execute() {
 	var (
-		longHelpText = map[string]string{}
-		app          = kingpin.New("amtool", "Alertmanager CLI").DefaultEnvars()
+		app = kingpin.New("amtool", helpRoot).DefaultEnvars()
```

Comment:
I think `DefaultEnvars()` can be removed, right?

Reply:
k. I doubt anyone is relying on it, but I suppose if someone is we can leave it for now.